deploy/charts/openshift-routes/templates/_helpers.tpl

```diff
@@ -48,18 +48,27 @@ https://github.com/helm/helm/issues/5358.
 Common labels
 */}}
 {{- define "openshift-routes.labels" -}}
-helm.sh/chart: {{ include "openshift-routes.chart" . }}
-{{ include "openshift-routes.selectorLabels" . }}
-app.kubernetes.io/component: controller
+{{/*
+You can generate generate a static manifest free of Helm-specific labels by
+using `--set omitHelmLabels=true`.
+*/}}
+{{- include "openshift-routes.selectorLabels" . }}
+{{- if not .Values.omitHelmLabels }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+helm.sh/chart: {{ include "openshift-routes.chart" . }}
+{{- end }}
 {{- end }}
 
 {{/*
 Selector labels
 */}}
 {{- define "openshift-routes.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "openshift-routes.name" . }}
+app.kubernetes.io/component: controller
+app.kubernetes.io/version: {{ .Chart.AppVersion }}
```

Comment:
@maelvls I don't think we should use the version as a selector label.
I expect that this will cause weirdness when upgrading from one version to the next (pods will no longer match the replicaset).

Reply:
Looks like you already noticed this: https://github.com/cert-manager/openshift-routes/pull/74#issuecomment-2238828278

Reply:
I added a commit to this PR to remove the version from the selector labels: https://github.com/cert-manager/openshift-routes/pull/74/commits/b36d5372ac760ee3a70414e02ce7abdcc3b0d0d1